### ISSUE TITLE
S3: GetPreSignedUrlRequest SignatureDoesNotMatch exception when using an object key that has a path separator

### ISSUE THREAD

**Issue by swijnands** [aws/aws-sdk-net#244]
Nuget package versions: 
 Core 3.1.1.1
 S3 3.1.2.1

Trying to presign a url for object keys with a directory separator '/' in them results in a SignatureDoesNotMatch exception.

Example code:

```csharp
var request = new PutObjectRequest
{
    BucketName = BucketName,
    Key = "people-dev/fa91d071-b9e6-4962-8e4d-5f4f4a6e0f64.jpg",
    ContentType = contentType,
    InputStream = stream
};

var response = await Client.PutObjectAsync(request);

var urlRequest = new GetPreSignedUrlRequest
{
    BucketName = request.BucketName,
    Key = request.Key,
    Expires = DateTime.UtcNow.AddDays(6)
};

return Client.GetPreSignedURL(urlRequest);
```

The PutObjectRequest works without problem.
However the trying to access the presigned url I get the following error:

```xml
<Error>
   <Code>SignatureDoesNotMatch</Code>
   <Message>The request signature we calculated does not match the signature you provided. Check your key and signing method.</Message>
   ....
</Error>
```

Crudely hex-escaping the slash character (request.Key.Replace("/", "%2F")) in an object key during a GetPreSignedUrlRequest, sometimes works, sometimes doesn't.

(The %2F becomes %252F in the final request url).

**Comment by PavelSafronov** [comment]
You shouldn't have to escape the slash character manually.
How are you consuming the presigned url? Is it in the browser, with code, etc.? We tried reproducing this issue with the key you mentioned, accessing the presigned url through code and with a number of browsers, and were unable to get the SignatureDoesNotMatch error.

**Comment by swijnands** [comment]
Thanks for your response!

I was consuming them using the browser (Chrome).
I've looked back in my chrome history to see if the urls where mangled, but that was not the case.
However, I saw that I was erroneously setting the ContentType attribute (passing it along from the upload request) on a GetPresignedURLRequest with a standard GET http verb.

So, that was the reason why I was getting a SignatureDoesNotMatch error!
